crates/ide_assists/src/handlers/replace_if_let_with_match.rs

```diff
@@ -235,12 +235,14 @@ fn pick_pattern_and_expr_order(
 ) -> Option<(ast::Pat, ast::Expr, ast::Expr)> {
     let res = match (pat, pat2) {
         (ast::Pat::WildcardPat(_), _) => return None,
-        (pat, sad_pat) if is_sad_pat(sema, &sad_pat) => (pat, expr, expr2),
-        (sad_pat, pat) if is_sad_pat(sema, &sad_pat) => (pat, expr2, expr),
+        (pat, _) if expr2.syntax().first_child().is_none() => (pat, expr, expr2),
+        (_, pat) if expr.syntax().first_child().is_none() => (pat, expr2, expr),
```

Comment:
I *think* this fails for literal expressions like numbers and the like.

I'd prefer if we checked the expr for the unit expression `()` and an empty block expression `{}` instead.

Reply:
I'm not too familiar with the Ast API used here, how'd i best check for empty blocks?
Do i directly pattern-match on the pattern type? Thanks for you help

Reply:
yep you would match against the `ast::Expr` enum variants and do some extra checks for tuple and block expressions.
For block expressions you will only need to check that the `statements()` iter on it is empty as well as that `tail_expr()` is none. For the tuple expression you just wanna check its `fields()` iter is empty.

Reply:
Updated 👍🏻